SingularityUI/app/components/requestDetail/header/RequestTitle.jsx

```diff
@@ -16,40 +17,67 @@ const errorDescription = (requestAPI) => {
   }
 };
 
-const RequestTitle = ({requestId, requestAPI}) => {
-  let maybeInfo;
-  if (Utils.api.isFirstLoad(requestAPI)) {
-    maybeInfo = <em>Loading...</em>;
-  } else if (requestAPI.error) {
-    const errorText = errorDescription(requestAPI);
-    maybeInfo = <p className="text-danger">{requestAPI.statusCode}: {errorText}</p>;
-  } else {
-    const requestParent = requestAPI.data;
-    const {request, state} = requestParent;
-    maybeInfo = (
-      <span>
-        <RequestStar requestId={request.id} />
-        <span className="request-state" data-state={state}>
-          {Utils.humanizeText(state)}
-        </span>
-        <span className="request-type">
-          {Utils.humanizeText(request.requestType)}
+class RequestTitle extends Component {
+  static propTypes = {
+    requestId: PropTypes.string.isRequired,
+    requestAPI: PropTypes.object
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {hover: false};
+  }
+
+  onMouseOver () {
+    this.setState({ hover: true });
+  }
+
+  render() {
+    const {requestAPI, requestId} = this.props;
+    const {hover} = this.state;
+    let maybeInfo;
+    if (Utils.api.isFirstLoad(requestAPI)) {
+      maybeInfo = <em>Loading...</em>;
+    } else if (requestAPI.error) {
+      const errorText = errorDescription(requestAPI);
+      maybeInfo = <p className="text-danger">{requestAPI.statusCode}: {errorText}</p>;
+    } else {
+      const requestParent = requestAPI.data;
+      const {request, state} = requestParent;
+      maybeInfo = (
+        <span>
+          <RequestStar requestId={request.id} />
+          <span className="request-state" data-state={state}>
+            {Utils.humanizeText(state)}
+          </span>
+          <span className="request-type">
+            {Utils.humanizeText(request.requestType)}
+          </span>
         </span>
-      </span>
+      );
+    }
+
+    const requestIdToDisplay = Utils.maybe(requestAPI, ['data', 'request', 'id']) || requestId;
+
+    return (
+      <div onMouseOver={() => this.onMouseOver()}>
```

Comment:
If `this.onMouseOver` is bound, you can just do `onMouseOver={this.onMouseOver}`